Backend/CourseService.cs

```diff
@@ -55,19 +55,18 @@ public List<Course> FindByStartName(string courseStartsWith)
         }
 
         /// <summary>
-        /// If course with "courseName" exists function return ID, if course doesnt exist fuction
-        /// save this course and return id of saved course.
+        /// If course with exists, function returns ID. If course doesnt exist function
+        /// saves this course and returns id of saved course.
         /// </summary>
-        public async Task<Course> GetOrAddNotExistsCourseByShortcut(string courseShortcut)
+        public async Task<Course> GetOrAddNotExistsCourse(string courseShortcut, string courseName)
         {
-            var course = await this.FindByCodeAsync(courseShortcut);
+            var course = await (string.IsNullOrEmpty(courseShortcut) ? this.FindByCodeAsync(courseName) : this.FindByCodeAsync(courseShortcut));
```

Comment:
Is this OK? you call FindByCodeAsync 2x, should not be one of them FindbyName?

Reply:
FindByCodeAsync is called only once depending on condition string.IsNullOrEmpty(courseShortcut). If string.IsNullOrEmpty(courseShortcut) is true then this.FindByCodeAsync(**courseName**) is called otherwise this.FindByCodeAsync(**courseShortcut**)

Reply:
:-) the same method just with differnet string? It has to be magic function :-)
```
        public async Task<Course> FindByCodeAsync(string code)
        {
            return await _courseCollection.Find(x => x.CourseCode.Equals(code)).FirstOrDefaultAsync();
        }
        public async Task<Course> FindByNameAsync(string name)
        {
            return await _courseCollection.Find(x => x.CourseName.Equals(name.ToLower())).FirstOrDefaultAsync();
        }
```

Reply:
Sry, didnt see that. Now it is corrected :)